### PR TITLE
Fix #9352: Unexpected error when certificate file does not exist

### LINKED ISSUE #9352: az keyvault certificate import fails with an unexpected error when file does not exist
**Describe the bug**
`az keyvault certificate import` fails with an unexpected error (including stack trace) when the file to be imported does not exist. It should return with a normal error stating that the file does not exist, similar to the error that is reported when the file does exist but is not a valid certificate file.

**To Reproduce**
```
$ az keyvault certificate import --vault-name doesnotexist --name doesnotexist --file doesnotexist
The command failed with an unexpected error. Here is the traceback:

[Errno 2] No such file or directory: 'doesnotexist'
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python2.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 507, in execute
    parsed_args = self.parser.parse_args(args)
  File "/usr/lib64/az/lib/python2.7/site-packages/knack/parser.py", line 256, in parse_args
    return super(CLICommandParser, self).parse_args(args)
  File "/usr/lib64/python2.7/argparse.py", line 1688, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1720, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1908, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/usr/lib64/python2.7/argparse.py", line 1885, in consume_positionals
    take_action(action, args)
  File "/usr/lib64/python2.7/argparse.py", line 1794, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib64/python2.7/argparse.py", line 1090, in __call__
    namespace, arg_strings = parser.parse_known_args(arg_strings, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1720, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1908, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/usr/lib64/python2.7/argparse.py", line 1885, in consume_positionals
    take_action(action, args)
  File "/usr/lib64/python2.7/argparse.py", line 1794, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib64/python2.7/argparse.py", line 1090, in __call__
    namespace, arg_strings = parser.parse_known_args(arg_strings, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1720, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1908, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/usr/lib64/python2.7/argparse.py", line 1885, in consume_positionals
    take_action(action, args)
  File "/usr/lib64/python2.7/argparse.py", line 1794, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib64/python2.7/argparse.py", line 1090, in __call__
    namespace, arg_strings = parser.parse_known_args(arg_strings, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1720, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python2.7/argparse.py", line 1926, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib64/python2.7/argparse.py", line 1866, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib64/python2.7/argparse.py", line 1778, in take_action
    argument_values = self._get_values(action, argument_strings)
  File "/usr/lib64/python2.7/argparse.py", line 2218, in _get_values
    value = self._get_value(action, arg_string)
  File "/usr/lib64/python2.7/argparse.py", line 2247, in _get_value
    result = type_func(arg_string)
  File "/usr/lib64/az/lib/python2.7/site-packages/azure/cli/command_modules/keyvault/_validators.py", line 196, in certificate_type
    with open(os.path.expanduser(string), 'rb') as f:
IOError: [Errno 2] No such file or directory: 'doesnotexist'

To open an issue, please run: 'az feedback'
$ touch doesnotexist
$ az keyvault certificate import --vault-name notmyvault --name notmyvault --file doesnotexist
We could not parse the provided certificate as .pem or .pfx. Please verify the certificate with OpenSSL.
$
```

**Expected behavior**
A normal error message that the file does not exist, e.g.:
```
$ az keyvault certificate import --vault-name notmyvault --name notmyvault --file example.pfx
The certificate file does not exist.
```

**Environment summary**
```
$ az --version
azure-cli                         2.0.64

acr                                2.2.6
acs                                2.4.1
advisor                            2.0.0
ams                                0.4.5
appservice                        0.2.19
backup                             1.2.4
batch                              4.0.1
batchai                            0.4.8
billing                            0.2.1
botservice                         0.2.0
cdn                                0.2.3
cloud                              2.1.1
cognitiveservices                  0.2.5
command-modules-nspkg               2.0.2
configure                         2.0.23
consumption                        0.4.2
container                         0.3.16
core                              2.0.64
cosmosdb                          0.2.10
deploymentmanager                  0.1.0
dla                                0.2.5
dls                                0.1.9
dms                                0.1.3
eventgrid                          0.2.3
eventhubs                          0.3.5
extension                          0.2.5
feedback                           2.2.1
find                               0.3.2
hdinsight                          0.3.3
interactive                        0.4.3
iot                                0.3.8
iotcentral                         0.1.6
keyvault                          2.2.15
kusto                              0.2.2
lab                                0.1.7
maps                               0.3.4
monitor                           0.2.13
network                            2.4.0
nspkg                              3.0.3
policyinsights                     0.1.3
privatedns                         1.0.0
profile                            2.1.5
rdbms                             0.3.10
redis                              0.4.2
relay                              0.1.4
reservations                       0.4.2
resource                          2.1.14
role                               2.6.1
search                             0.1.1
security                           0.1.1
servicebus                         0.3.5
servicefabric                     0.1.18
signalr                            1.0.0
sql                                2.2.3
sqlvm                              0.1.1
storage                            2.4.1
telemetry                          1.0.2
vm                                2.2.20

Python location '/usr/lib64/az/bin/python'
Extensions directory '/home/oskamj/.azure/cliextensions'

Python (Linux) 2.7.5 (default, Apr  9 2019, 14:30:50)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
$ uname -a
Linux jurjenoskame002 3.10.0-957.12.1.el7.x86_64 #1 SMP Mon Apr 29 14:59:59 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)
$
```

### PR DESCRIPTION
**Pull request by jiasli**
Fix #9352: Unexpected error when certificate file does not exist
Tweak logging level to appear with --verbose
Fix typo

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/_validators.py [hunk 1]
```diff
@@ -203,9 +204,14 @@ def _load_certificate_as_bytes(file_name):
...
-        cert_data = f.read()
-    return cert_data
+    try:
+        with open(os.path.expanduser(string), 'rb') as f:
+            cert_data = f.read()
+        return cert_data
+    except (IOError, OSError) as e:  # FileNotFoundError introduced in Python 3
+        if e.errno == errno.ENOENT:
+            raise CLIError("Certificate file '{}' does not exist.".format(string))
+        raise CLIError("Unable to load certificate file '{}'.".format(string))
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/keyvault/_validators.py]: I suggest include the error detail

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def _maybe_load_file(arg):
             try:
                 return _load_file(poss_file)
             except IOError:
-                logger.debug("Failed to load @'%s', assume not a file", arg)
+                logger.info("Failed to load '%s', assume not a file", arg)
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: Mentioning the `@` is on purpose, hence I suggest keep it.

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/commands/__init__.py]: `arg` already contains `@` (`arg.find('@')`), the original code will print duplicated `@`.

File: src/azure-cli/azure/cli/command_modules/keyvault/_validators.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/_validators.py]: Changed to use exception's own error message

**Comment by yonzhan** [comment]
@jiasli please help with this PR.

**Comment by haroldrandom** [comment]
@jiasli @yugangw-msft Have you come to an agreement to this PR?

File: src/azure-cli-core/azure/cli/core/commands/__init__.py (hunk 2 above)
Comment (haroldrandom) [reply src/azure-cli-core/azure/cli/core/commands/__init__.py]: I think a spare `@` is not necessary.

**Pull request merged**